om4labs/diags/stress_curl/stress_curl.py

```diff
@@ -0,0 +1,259 @@
+
+
+def read(dictArgs):
+    ds_model = xr.open_mfdataset(dictArgs["infile"], use_cftime=True)
+    dates    = date_range(ds_model)
+    ds_ref_tau = xr.open_mfdataset(
+        [dictArgs["ref_taux"], dictArgs["ref_tauy"]], use_cftime=True
+    )
+    ds_static = xr.open_mfdataset(dictArgs["static"])
+
+    # replace the nominal xq and yq by indices so that Xarray does not get confused.
+    # Confusion arises since there are inconsistencies between static file grid and
+    # model data grid for the last value of yq. We never need xq and yq for actual
+    # calculations, so filling these arrays with any value is not going to change
+    # any results. But Xarray needs them to be consistent between the two files when
+    # doing the curl operation on the stress.
+    ds_model["xq"] = xr.DataArray(np.arange(len(ds_model["xq"])), dims=["xq"])
+    ds_model["yq"] = xr.DataArray(np.arange(len(ds_model["yq"])), dims=["yq"])    
+    ds_ref_tau["xq"] = xr.DataArray(np.arange(len(ds_ref_tau["xq"])), dims=["xq"])
+    ds_ref_tau["yq"] = xr.DataArray(np.arange(len(ds_ref_tau["yq"])), dims=["yq"])
+    ds_static["xq"] = xr.DataArray(np.arange(len(ds_static["xq"])), dims=["xq"])
+    ds_static["yq"] = xr.DataArray(np.arange(len(ds_static["yq"])), dims=["yq"])
+
+    ds_model.attrs = {"date_range":dates}
+
+    return ds_model, ds_ref_tau, ds_static
+
+def calc_curl_stress(
+    ds,
+    ds_static,
+    varx="tauuo",
+    vary="tauvo",
+    areacello_bu="areacello_bu",
+    xdim="lon",
+    ydim="lat",
+):
+    """Calculate curl of stress acting on surface of the ocean.
+
+    Parameters
+        ----------
+    ds        : xarray.Dataset dataset with tauuo and tauvo
+    ds_static : xarray.Dataset with grid values
+    varname   : str, optional
+        Name of the tauuo and tauvo variables, by default "tauuo" and "tauvo"
+    area : str, optional
+        Name of the area variable, by default "areacello_bu"
+    xdim : str, optional
+        Name of the longitude coordinate, by default "lon"
+    ydim : str, optional
+        Name of the latitude coordinate, by default "lat"
+    Returns
+    -------
+        xarray.DataArray stress_curl
+        curl of surface ocean stress
+    """
+
+    rho0 = 1035.0
+    area = ds_static[areacello_bu]
+    taux = ds[varx]
+    taux = taux.mean(dim="time")
+    tauy = ds[vary]
+    tauy = tauy.mean(dim="time")
+
+    # fill nan with 0.0 since want 0.0 values over land for the curl operation
+    taux = taux.fillna(0.0)
+    tauy = tauy.fillna(0.0)
+
+    grid = Grid(
+        ds_static,
+        coords={
+            "X": {"center": "xh", "outer": "xq"},
+            "Y": {"center": "yh", "outer": "yq"},
+        },
+        periodic=["X"],
+    )
+
+    stress_curl = -grid.diff(taux * ds_static.dxCu, "Y", boundary="fill") + grid.diff(
+        tauy * ds_static.dyCv, "X", boundary="fill"
+    )
+
+    stress_curl = stress_curl / (area * rho0)
+    stress_curl = stress_curl.where(ds_static["wet_c"] == 1)
```

Comment:
"wet_c", "geolon_c",... should be stored as default to kwargs in case someone has different naming convention